controllers/imageclusterinstall_controller.go

```diff
@@ -220,18 +220,28 @@ func (r *ImageClusterInstallReconciler) Reconcile(ctx context.Context, req ctrl.
 			return ctrl.Result{}, err
 		}
 	}
+	// This code was moved here to avoid updating the RequirementsMet condition back and forth
```

Comment:
If you want to keep this comment I'd avoid talking about how it has been "moved". This feels like it belongs more in the commit message

Maybe something like
```suggestion
	// validate the data image isn't being deleted before setting `RequirementsMet` to success in `validateBMH`
```